### ISSUE TITLE
Precision and accuracy of pose estimation and tracking

### ISSUE THREAD

**Issue by Hotsburp** [NVlabs/FoundationPose#144]
Thank you for the excellent work!

I have tried out the model-based version of FoundationPose and it works very well even on custom objects.

I would like to ask what is the estimated error in pose estimation (rotation and translation) generally when using FoundationPose based on the trained objects and general dataset?

For example, using a simple object like a box, the pose variable may return [x, y, z] for the translation component relative to the camera. But in reality, the actual translation position is a deviation of plus minus x', y', z' from x, y, z values respectively.

**Comment by wenbowen123** [comment]
Hi, it's usually within a few mm (also depends on a couple of things like the depth quality, image resolution, camera intrinsic calibration error etc).

**Comment by Hotsburp** [comment]
Thank you for the prompt response and insights!

**Comment by adennaoui97** [comment]
Hello. 
How do these parameters affect the precision? Higher resolution is better? Is the depth quality of RealSense D455 cameras good enough?

**Comment by wenbowen123** [comment]
@adennaoui97 higher resolution and more accurate depth definitely helps. For the latter, you can check our recent work [FoundationStereo](https://github.com/NVlabs/FoundationStereo)

**Comment by adennaoui97** [comment]
@wenbowen123 
Hi, I integrated FoundationStereo and FoundationPose to estimate the pose of the object in one frame - so pose estimation without tracking since FoundationStereo takes a bit to process a single frame so I did not even try the tracking yet (I was thinking that if the initial pose estimation is good, the tracking is continued by getting the depth measurements from the camera - in theory, worse than the output of FoundationStereo).

However, the end result was very similar to using FoundationPose without FoundationStereo, despite the depth output from FoundationStereo is visually way better. WOW.

However, I have a question - just to be sure that I am approaching the tools as I should.

In FoundationPose, camK.txt contains the RGB or depth intrinsics? (I use RealSense D455 and I access the intrinsics value using pyrealsense2 wrapper of librealsense).
In FoundationStereo, I give as an output both left and right infrared images and K.txt - I got that the second line contains the baseline distance between the left and right stereo cameras but again, the first line that contains the flattened intrinsics represent the intrinsics of RGB camera or left (or right) stereo camera (or the depth camera)?


I also need to point out that the depth output of FoundationStereo that is the result of both stereo cameras will be (and I am assuming here) aligned with the depth image that is obtained by RealSense directly but it will not be aligned with the RGB image (which I usually align to RGB camera using pyrealsense2.align.process()) so those who want to use FoundationStereo to obtain the depth image has to project the 2D depth image into a 3D depth space (using the depth intrinsics?),  then align the depth to the RGB image using the extrinsics (rotation matrix and translation vector, also obtained using the RealSense API), then go from 3D aligned depth space to 2D aligned depth image (using RGB intrinsics?), and since we are working with Numpy arrays - the shift successful but the depth image has some weird noises but interestingly, FoundationPose worked with that manually aligned depth image (and it did not work with unaligned output depth image, coming out of FoundationStereo).

TL; DR: I am unsure the intrinsics of which camera to use in FoundationPose, FoundationStereo, and in the aligning depth image with RGB image process.
